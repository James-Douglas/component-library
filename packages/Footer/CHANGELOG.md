# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.12.7](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.12.6...@comparethemarketau/manor-footer@0.12.7) (2021-04-12)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.12.6](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.12.5...@comparethemarketau/manor-footer@0.12.6) (2021-03-18)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.12.5](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.12.4...@comparethemarketau/manor-footer@0.12.5) (2021-03-09)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.12.4](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.12.3...@comparethemarketau/manor-footer@0.12.4) (2021-02-26)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.12.3](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.12.2...@comparethemarketau/manor-footer@0.12.3) (2021-02-23)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.12.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.12.1...@comparethemarketau/manor-footer@0.12.2) (2021-02-20)


### Bug Fixes

* **footer:** apply className to outermost div on footer ([#578](https://github.com/comparethemarketau/manor-react/issues/578)) ([16a19a5](https://github.com/comparethemarketau/manor-react/commit/16a19a5d5d59d230d7f8f9f1b3b435c6b60aea7d))





## [0.12.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.12.0...@comparethemarketau/manor-footer@0.12.1) (2021-02-09)

**Note:** Version bump only for package @comparethemarketau/manor-footer





# [0.12.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.11.9...@comparethemarketau/manor-footer@0.12.0) (2021-01-07)


### Features

* ðŸŽ¸ removed product cards ([#527](https://github.com/comparethemarketau/manor-react/issues/527)) ([681626a](https://github.com/comparethemarketau/manor-react/commit/681626a8fee4924aec50602fed22a23e48f7761d))


### BREAKING CHANGES

* ðŸ§¨ product cards removed from manor





## [0.11.9](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.11.8...@comparethemarketau/manor-footer@0.11.9) (2020-12-18)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.11.8](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.11.7...@comparethemarketau/manor-footer@0.11.8) (2020-12-16)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.11.7](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.11.6...@comparethemarketau/manor-footer@0.11.7) (2020-12-08)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.11.6](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.11.5...@comparethemarketau/manor-footer@0.11.6) (2020-11-30)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.11.5](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.11.4...@comparethemarketau/manor-footer@0.11.5) (2020-11-24)


### Bug Fixes

* **footer:** updated footer acn formatting ([#499](https://github.com/comparethemarketau/manor-react/issues/499)) ([fe8b306](https://github.com/comparethemarketau/manor-react/commit/fe8b3068bd77d9d26dc084a07786a783056b1db1))





## [0.11.4](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.11.3...@comparethemarketau/manor-footer@0.11.4) (2020-11-23)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.11.3](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.11.2...@comparethemarketau/manor-footer@0.11.3) (2020-11-15)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.11.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.11.1...@comparethemarketau/manor-footer@0.11.2) (2020-11-12)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.11.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.11.0...@comparethemarketau/manor-footer@0.11.1) (2020-11-12)

**Note:** Version bump only for package @comparethemarketau/manor-footer





# [0.11.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.10.0...@comparethemarketau/manor-footer@0.11.0) (2020-11-09)


### Features

* **typography:** remove nomargins pro ([#470](https://github.com/comparethemarketau/manor-react/issues/470)) ([ad14285](https://github.com/comparethemarketau/manor-react/commit/ad14285b548116be2306e19bb989113397a11e3f))





# [0.10.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.9.3...@comparethemarketau/manor-footer@0.10.0) (2020-11-06)


### Features

* **typography:** added help text style ([#469](https://github.com/comparethemarketau/manor-react/issues/469)) ([0827882](https://github.com/comparethemarketau/manor-react/commit/0827882cd6020d5dbada7371faa19660c961af4e))





## [0.9.3](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.9.2...@comparethemarketau/manor-footer@0.9.3) (2020-10-22)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.9.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.9.1...@comparethemarketau/manor-footer@0.9.2) (2020-10-21)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.9.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.9.0...@comparethemarketau/manor-footer@0.9.1) (2020-10-20)

**Note:** Version bump only for package @comparethemarketau/manor-footer





# [0.9.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.8.3...@comparethemarketau/manor-footer@0.9.0) (2020-10-19)


### Features

* **typography:** rewrote typography to no longer use materialUI ([#431](https://github.com/comparethemarketau/manor-react/issues/431)) ([6a7f7db](https://github.com/comparethemarketau/manor-react/commit/6a7f7dbef8f8bea684655f598c2ec9d55b814c34))


### BREAKING CHANGES

* **typography:** props removed: paragraph, gutterBottom. prop changes: noMargins - no defaults to
true, variant - 'button', 'inherit' options removed

* feat(typography): sronly implemented

* feat(typography): address pr feedback

* feat(typography): default display prop to 'block' instead of 'initial'

* feat(typography): update components impacted by typography changes





## [0.8.3](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.8.2...@comparethemarketau/manor-footer@0.8.3) (2020-09-29)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.8.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.8.1...@comparethemarketau/manor-footer@0.8.2) (2020-09-29)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.8.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.8.0...@comparethemarketau/manor-footer@0.8.1) (2020-09-25)

**Note:** Version bump only for package @comparethemarketau/manor-footer





# [0.8.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.7.10...@comparethemarketau/manor-footer@0.8.0) (2020-09-24)


### Features

* **rollup:** update rollup to not bundle manor-provider ([#403](https://github.com/comparethemarketau/manor-react/issues/403)) ([0e012d6](https://github.com/comparethemarketau/manor-react/commit/0e012d6fbadcf0ec99857c22e148cacd6265b60a))





## [0.7.10](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.7.9...@comparethemarketau/manor-footer@0.7.10) (2020-09-21)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.7.9](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.7.8...@comparethemarketau/manor-footer@0.7.9) (2020-09-18)


### Bug Fixes

* **dependencies:** fix more incorrect dependencies ([#392](https://github.com/comparethemarketau/manor-react/issues/392)) ([6fa55a1](https://github.com/comparethemarketau/manor-react/commit/6fa55a11ba89125ccfe61385d9776e4185bff6f3))





## [0.7.8](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.7.7...@comparethemarketau/manor-footer@0.7.8) (2020-09-18)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.7.7](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.7.6...@comparethemarketau/manor-footer@0.7.7) (2020-09-16)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.7.6](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.7.5...@comparethemarketau/manor-footer@0.7.6) (2020-09-16)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.7.5](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.7.4...@comparethemarketau/manor-footer@0.7.5) (2020-09-09)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.7.4](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.7.3...@comparethemarketau/manor-footer@0.7.4) (2020-09-08)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.7.3](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.7.2...@comparethemarketau/manor-footer@0.7.3) (2020-09-04)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.7.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.7.1...@comparethemarketau/manor-footer@0.7.2) (2020-09-03)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.7.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.7.0...@comparethemarketau/manor-footer@0.7.1) (2020-09-01)

**Note:** Version bump only for package @comparethemarketau/manor-footer





# [0.7.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.6.1...@comparethemarketau/manor-footer@0.7.0) (2020-08-18)


### Features

* **all:** remove manorprovider from components, fix typography props ([#365](https://github.com/comparethemarketau/manor-react/issues/365)) ([9297cf7](https://github.com/comparethemarketau/manor-react/commit/9297cf72e8a7fe8762ec0dadf07d026aa88cbb44))





## [0.6.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.6.0...@comparethemarketau/manor-footer@0.6.1) (2020-08-18)

**Note:** Version bump only for package @comparethemarketau/manor-footer





# [0.6.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.5.2...@comparethemarketau/manor-footer@0.6.0) (2020-08-17)


* Fix/global styles micro ui (#359) ([bcf8ce9](https://github.com/comparethemarketau/manor-react/commit/bcf8ce92ba170a51113a4022728da22f47a6a768)), closes [#359](https://github.com/comparethemarketau/manor-react/issues/359)


### BREAKING CHANGES

* Global styles have been removed from ManorProvider, as such typography is not
styled by default. Import the Font component once in your application (or macro-ui) and use the
Typography component to render text, and the Link component to render links.

* fix(fonts): fix fonts dependencies

* fix(deps): fix provider dependencies

* fix(feedback): fixes based on PR feedback

Co-authored-by: PSJgit <psjj.dev@gmail.com>





## [0.5.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.5.1...@comparethemarketau/manor-footer@0.5.2) (2020-08-11)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.5.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.5.0...@comparethemarketau/manor-footer@0.5.1) (2020-08-11)

**Note:** Version bump only for package @comparethemarketau/manor-footer





# [0.5.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.4.2...@comparethemarketau/manor-footer@0.5.0) (2020-08-10)


### Features

* **grid:** added customizable padding and gutterWidth ([#353](https://github.com/comparethemarketau/manor-react/issues/353)) ([87f55b5](https://github.com/comparethemarketau/manor-react/commit/87f55b5977801135c875b821c5b569feecea8dc7))





## [0.4.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.4.1...@comparethemarketau/manor-footer@0.4.2) (2020-08-07)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.4.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.4.0...@comparethemarketau/manor-footer@0.4.1) (2020-07-28)

**Note:** Version bump only for package @comparethemarketau/manor-footer





# [0.4.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.3.0...@comparethemarketau/manor-footer@0.4.0) (2020-07-24)


### Features

* **footer:** update to footer for customer accounts ([#351](https://github.com/comparethemarketau/manor-react/issues/351)) ([20d4897](https://github.com/comparethemarketau/manor-react/commit/20d4897017f72154239815869f14f9b6bf163a91))





# [0.3.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.2.5...@comparethemarketau/manor-footer@0.3.0) (2020-07-20)


### Features

* **providers:** remove LayerEvent/Toast providers from manorprovider ([#348](https://github.com/comparethemarketau/manor-react/issues/348)) ([461d724](https://github.com/comparethemarketau/manor-react/commit/461d72498fca1aca9de0056a27d1a3d17a89ea77))


### BREAKING CHANGES

* **providers:** drawers and modals are no longer managed by manor, users will need to use the
zIndex prop (added to both) to do so particularly when multiple layered components are rendered
simultaneously. Additionally, you will need to implement your own keyboard bindings and focus
trapping to ensure conformance to the design system.





## [0.2.5](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.2.4...@comparethemarketau/manor-footer@0.2.5) (2020-07-14)


### Bug Fixes

* **toastmanager:** removed portals from ToastManager ([#345](https://github.com/comparethemarketau/manor-react/issues/345)) ([1533377](https://github.com/comparethemarketau/manor-react/commit/1533377910e9cbac266abe24fae1ee42eba4c52f))





## [0.2.4](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.2.3...@comparethemarketau/manor-footer@0.2.4) (2020-06-21)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.2.3](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.2.2...@comparethemarketau/manor-footer@0.2.3) (2020-06-17)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.2.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.2.1...@comparethemarketau/manor-footer@0.2.2) (2020-06-16)

**Note:** Version bump only for package @comparethemarketau/manor-footer





## [0.2.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.2.0...@comparethemarketau/manor-footer@0.2.1) (2020-06-15)

**Note:** Version bump only for package @comparethemarketau/manor-footer





# [0.2.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-footer@0.1.0...@comparethemarketau/manor-footer@0.2.0) (2020-06-12)


### Features

* **toasts:** manorProvider now injects ToastManager ([#333](https://github.com/comparethemarketau/manor-react/issues/333)) ([8621888](https://github.com/comparethemarketau/manor-react/commit/862188867bbc8258b29fa162f46e5ad5b108f778))


### BREAKING CHANGES

* **toasts:** Added ToastNotification component. Refactored toasts such that ManorProvider
injects the ToastManager. addToasts also now expects a <ToastNotification> instead of an object.

* fix(story): fix typo in toast notification story





# 0.1.0 (2020-06-11)


### Features

* **packages:** migrated to monorepo ([920a45e](https://github.com/comparethemarketau/manor-react/commit/920a45ec4b40a19de32f39f29693cbe1b1f314ae))
