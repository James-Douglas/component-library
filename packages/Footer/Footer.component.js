import React, { useRef, useState, useLayoutEffect } from 'react';
import PropTypes from 'prop-types';
import { Column, Row, Container } from '@comparethemarketau/manor-grid';
import { Microcopy } from '@comparethemarketau/manor-typography';
import { throttle } from '@comparethemarketau/manor-utils';

import {
  StyledFooterBar, StyledP, StyledPosition, StyledWrapper,
} from './Footer.styles';

const Footer = ({
  children, background, sticky, type, className,
}) => {
  const currentYear = new Date().getFullYear();
  const footerRef = useRef(null);

  const [windowWidth, setWindowWidth] = useState();
  const [footerHeight, setFooterHeight] = useState();

  const copyRightText = 'Compare The Market. All rights reserved. ACN: 117323 378 AFSL 422926';

  // switch here to add more cases if necessary
  const renderVariant = () => {
    switch (type) {
      case 'customer-accounts':
        return (
          <>
            <StyledP>&copy; {currentYear} {copyRightText}</StyledP>
            {children && <Microcopy>{children}</Microcopy>}
          </>
        );
      default:
        return (
          <>
            {children && <Microcopy>{children}</Microcopy>}
            <StyledP>&copy; {currentYear} {copyRightText}</StyledP>
          </>
        );
    }
  };

  useLayoutEffect(() => {
    if (sticky) {
      const handleResize = () => setWindowWidth(window.innerWidth);
      const throttledResize = throttle(handleResize, 25);
      setFooterHeight(footerRef.current.offsetHeight);
      window.addEventListener('resize', throttledResize);
      return () => {
        window.removeEventListener('resize', throttledResize);
      };
    }
    return null;
  }, [sticky, windowWidth]);

  return (
    <StyledWrapper style={{ paddingTop: footerHeight }}>
      <StyledPosition background={background} sticky={sticky}>
        <Container className={className}>
          <Row removeMarginBottom>
            <Column cols={12}>
              <StyledFooterBar ref={footerRef}>
                {renderVariant()}
              </StyledFooterBar>
            </Column>
          </Row>
        </Container>
      </StyledPosition>
    </StyledWrapper>
  );
};

Footer.propTypes = {
  /**
   * The children of the footer component (text and links, microcopy)
   */
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]),
  /**
   * White background on, or set to transparent
   */
  background: PropTypes.bool,
  /**
   * Set the footer to stick to the bottom of the page, regardless of content
   */
  sticky: PropTypes.bool,
  /**
   * Type of footer, currently just the default and customer-accounts
   */
  type: PropTypes.oneOf(['default', 'customer-accounts']),
  /**
   * Classes to be applied to the Footer component
   */
  className: PropTypes.string,
};

Footer.defaultProps = {
  children: '',
  background: true,
  sticky: true,
  type: 'default',
  className: '',
};

export default Footer;
