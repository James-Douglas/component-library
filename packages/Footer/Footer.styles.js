import styled, { css } from 'styled-components';

export const StyledWrapper = styled.div`
  *, *:before, *:after {
    box-sizing: border-box;
  }
  span {
    font-size: ${({ theme }) => theme.fontSize.sm};
  }
  width: 100%;
`;

export const StyledPosition = styled.div`
  ${({ sticky }) => sticky
    && css`
      position: absolute;
      right: 0;
      bottom: 0;
      left: 0;
      width: 100%;
    `}
  background: ${({ theme, background }) => (background ? theme.footer.background : theme.footer.transparent)};  
`;

export const StyledFooterBar = styled.div`
  min-height: 5.4rem;
  width: 100%;
  text-align: center;
  padding: ${({ theme }) => theme.spacing[24]};
`;

export const StyledCustomerAccounts = styled.div`
  text-align: left;
  p, a {
    font-size: ${({ theme }) => theme.fontSize.sm};
  }
  a {
    text-decoration: none;
  }
`;

export const StyledMicrocopy = styled.div`
  span {
    display: block;
    font-family: ${({ theme }) => theme.fontFamily};
    font-weight: ${({ theme }) => theme.fontWeight.normal};
    font-size: ${({ theme }) => theme.fontSize.xs};
    line-height: ${({ theme }) => theme.lineHeight.snug};
    margin-bottom: ${({ theme }) => theme.spacing[12]}
  }
`;
