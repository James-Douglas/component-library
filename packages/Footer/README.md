# Footer

## Installation

`yarn add @comparethemarketau/manor-footer`


## Documentation

For technical documentation, see [Storybook](https://services.dev.comparethemarket.cloud/manor/?path=/docs/components-footer--footer-basic).
Design system documentation can be found here [Bento](https://zeroheight.com/9942937b5/p/984ef2-page-footer/b/97d6bd)
