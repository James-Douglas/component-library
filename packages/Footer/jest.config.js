const baseConfig = require('../../jest.config.base');

module.exports = {
  ...baseConfig,
  roots: [
    '<rootDir>/packages/Footer',
  ],
  collectCoverageFrom: [
    '*.{js,jsx,ts,tsx}',
  ],
  name: 'Footer',
  displayName: 'Footer',
  rootDir: '../..',
};
