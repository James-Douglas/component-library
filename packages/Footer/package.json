{
  "name": "@comparethemarketau/manor-footer",
  "version": "0.7.3",
  "description": "Manor Footer component",
  "main": "lib/index.cjs.js",
  "author": "Compare The Market",
  "license": "Proprietary",
  "scripts": {
    "build": "rollup -c '../../rollup.config.js'",
    "tdd": "jest --coverage --watchAll",
    "test": "cross-env JEST_SILENT_REPORTER_DOTS=true jest --reporters=jest-silent-reporter --coverage --watchAll",
    "test:ci": "cross-env CI=true jest",
    "prepare": "cp package.json lib/ && cp README.md lib/ && cp CHANGELOG.md lib/"
  },
  "publishConfig": {
    "registry": "https://nexus.comparethemarket.com.au/repository/platform-packages/",
    "access": "public",
    "tag": "latest"
  },
  "dependencies": {
    "@comparethemarketau/manor-grid": "^0.6.3",
    "@comparethemarketau/manor-typography": "^0.5.2",
    "@comparethemarketau/manor-utils": "^0.1.9",
    "prop-types": "^15.7.2"
  },
  "peerDependencies": {
    "react": "^16.11.0",
    "react-dom": "^16.11.0",
    "styled-components": "^5.0.0"
  }
}
