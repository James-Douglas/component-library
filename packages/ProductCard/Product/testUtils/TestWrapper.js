import React from 'react';
import ProductProvider from '../../Context/ProductProvider';

const ProductTestWrapper = ({
  // eslint-disable-next-line react/prop-types
  children, product, onDetails = () => {}, onSelect = () => {},
}) => (
  <ProductProvider product={product} onDetails={onDetails} onSelect={onSelect}>
    { children }
  </ProductProvider>
);

export default ProductTestWrapper;
