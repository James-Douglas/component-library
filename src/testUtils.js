// test-utils.js
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './providers/ManorProvider';

const customRender = (ui, options) => render(ui, { wrapper: ManorProvider, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
