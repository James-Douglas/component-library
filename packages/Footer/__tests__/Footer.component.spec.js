import React from 'react';
import { render } from '../../../testUtils';

import Footer from '../Footer.component';

describe('Footer', () => {
  it('renders with minimal props', () => {
    const { container } = render(<Footer />);
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('renders with disclaimer text and clickable links', () => {
    const { getByText, container } = render(
      <Footer>
        <>test disclaimer and a <a href="https://dev.comparethemarket.com.au/" id="link">link</a></>
      </Footer>,
    );
    const link = container.querySelector('#link');
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute('href', 'https://dev.comparethemarket.com.au/');
    expect(getByText('test disclaimer and a')).toBeInTheDocument();
  });

  it('renders with a white bg by default', () => {
    const { container } = render(
      <Footer>
        <>test disclaimer</>
      </Footer>,
    );
    const footer = container.firstChild.firstChild;
    expect(footer).toHaveStyle('background: #ffffff');
  });

  it('renders with a transparent bg when the prop is supplied', () => {
    const { container } = render(
      <Footer background={false}>
        <>test disclaimer</>
      </Footer>,
    );
    const footer = container.firstChild.firstChild;
    expect(footer).toHaveStyle('background: transparent');
  });
});
