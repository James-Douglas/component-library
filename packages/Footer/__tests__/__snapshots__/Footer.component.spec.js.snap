// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders with minimal props 1`] = `
"<div class=\\"sc-fznZeY iscnKp\\" style=\\"padding-top: 0px;\\"><div class=\\"sc-fzokOt jUStbm\\"><div class=\\"sc-AxheI iNVRxu\\"><div class=\\"sc-AxhCb sc-AxmLO cPhYWV row \\"><div class=\\"sc-AxhUy sc-AxgMl gzcspF 
        
      \\"><div class=\\"sc-fzqBZW jARrmz\\"><p class=\\"sc-fzqNJr iUCRId\\">Â© 2020 Compare The Market. All rights reserved. ACN: 117323 378 AFSL 422926</p></div></div></div></div></div></div>"
`;
